gateway/handle_http_auth.go

```diff
@@ -119,6 +123,57 @@ func (gw *Handle) sourceIDAuth(delegate http.HandlerFunc) http.HandlerFunc {
 	}
 }
 
+// authDestIDForSource middleware to authenticate destinationId in the X-Rudder-Destination-Id header.
+// If the destinationId is invalid, the request is rejected.
```

Comment:
Do we really need to validate destinationId at the gateway level? The requests will anyways be filtered out in processor and since it is only an internal endpoint don't think sources would send an invalid destinationId. We could just accept whatever we get in the gateway level and move on

Reply:
IMO we should not trust what's coming in requests, we should always validate critical inputs.